department-of-veterans-affairs/caseflow
# Kamalam7/appeals 13374

Resolves APPEALS-13374

Description

Create Resend final notification letter from Send Final Notification Letter Task. In the Clerk of the board view the Send Final Notification Letter task  cases are under Unassigned tab.
If you click on the Case details, in the Currently active task you can see the task as Send Final Notification Letter. In the Currently active task, from Actions dropdown selection you need to select Resend final notification letter. When you click on the case and go to Case details you can see the task as Send Final Notification Letter. In the Case timeline you can see the Send Final Notification Letter Completed.

Acceptance Criteria
- [x] The following occurs when the "Resent final notification letter" task action is selected 
        for the Send final notification letter task
              Modal appears requiring user to confirm intent
              User is required to provide context
                    field type: free text 350 character limit
- [x] Once confirmed:
               Send final notification letter task is marked as complete
               Task moves from Currently Active Tasks section to Case Timeline and contains 
               the following information
                      Date of occurrence
                      Assigned to: <user or organization>
                      Assigned by: <user who completed action to assign task to COB>
                      Completed by: <user who completed Send Final Notification Letter task>
                      Task Instructions: <user input in the Resent initial notification letter modal>
- [x] A NEW Send Final Notification task from is created and active
              Task appears within the Currently Active Tasks section
              Task is assigned to the Clerk of the Board organization 
              Case appears in the unassigned tab of the Clerk of the Board organizational 
                queue


Testing Plan

1. Go to Caseflow Login and Sign as BVADWISE
2. Navigate to Mail Intake
3. From the dropdown select VA FORM 10182 for processing and click on continue to 
    search
4. Enter a veteran file number. I used 711790014. Click the search icon to continue
5. Fill out the form and select direct review for the review option type. Click the button at 
    the bottom to continue
6. On the add issues page, click the add issues button. Select a benefit type as 
    Compensation and Issue Category as "Contested claims - Attorney Fees. Click on Add 
     this issue and establish appeal to complete the intake
7. The intake should be completed
8. Switch user and select Clerk of the board user and enter into Queue and click on Switch 
    view and select Clerk of the board team view
<img width="836" alt="image" src="https://user-images.githubusercontent.com/110078646/222497028-72f01e49-416b-4174-9205-508995b88592.png">

9. The  Send Final Notification Letter task is created in 2 ways. One is from Send Initial Notification Letter Task and the other 
     one is from Post Send Initial Notification Letter Holding Task. 
10. From the Send Initial Notification Letter task you need to select and click on Mark as complete or Proceed to final letter.
      If you select Mark as complete, the Task will be created as Post Send Initial Notification Letter Holding Task.
      And the case is moved from "Unassigned tab" to "On hold tab" 
      From Post Send Initial Notification Letter Holding Task you can select Proceed to final notification letter.
      Now the Send Final Notification Letter Task is created and the case moved from "On hold tab" to "Unassigned tab"

<img width="832" alt="image" src="https://user-images.githubusercontent.com/110078646/222504350-4cddf136-7a51-45cf-bdbf-c6b64d81314d.png">

<img width="833" alt="image" src="https://user-images.githubusercontent.com/110078646/222504457-cc61212d-4f94-4e9d-8378-4ef9e3cd6cc5.png">

<img width="852" alt="image" src="https://user-images.githubusercontent.com/110078646/222504553-61143864-84b8-482d-b497-8962f06dc2e9.png">

<img width="839" alt="image" src="https://user-images.githubusercontent.com/110078646/222504682-7b3bba54-c9d4-451d-a8cb-d4a3def259dd.png">

<img width="830" alt="image" src="https://user-images.githubusercontent.com/110078646/222504843-897f7e1c-54bb-4924-80cf-9c1e4dcd836f.png">

<img width="837" alt="image" src="https://user-images.githubusercontent.com/110078646/222504945-6e607d46-04d6-452d-92bb-547bc088dc4a.png">

<img width="826" alt="image" src="https://user-images.githubusercontent.com/110078646/222505245-5fed6b1b-752a-4ffe-a58a-31e14aac74e5.png">

11. Now in the Currently active task you can see the Send Final Notification Letter task is created. From the actions dropdown 
      you can select "Resend Final Notification Letter" 

<img width="802" alt="image" src="https://user-images.githubusercontent.com/110078646/222506249-807fb2a5-b0ed-4799-b748-0b9a6ba302ed.png">

<img width="818" alt="image" src="https://user-images.githubusercontent.com/110078646/222506341-15d6c5d8-f7a3-4247-bb3f-8e098aa48444.png">

<img width="830" alt="image" src="https://user-images.githubusercontent.com/110078646/222506607-2f596d75-d9e6-4552-ae1b-b44bc16c6fd7.png">

12. If you go to the case details you can see the Task as Send Final Notification Letter.
<img width="818" alt="image" src="https://user-images.githubusercontent.com/110078646/222507129-43f5481e-d44e-447d-8efa-5112a8f9cb15.png">

13. If you go the Case timeline you can see all the completed tasks

<img width="830" alt="image" src="https://user-images.githubusercontent.com/110078646/222507322-78b01f8e-e091-41ac-a408-ca49c70be9a3.png">

14. From the Send Initial Notification Letter, if you select and click on Proceed to final notification letter the task will be created as send Final Notification Letter. (Here I picked another case to go with the Contested Claim workflow)

<img width="794" alt="image" src="https://user-images.githubusercontent.com/110078646/222509095-7c3b43fb-3f48-434f-8f22-cb991d607b46.png">

<img width="822" alt="image" src="https://user-images.githubusercontent.com/110078646/222509303-7d9c2c8b-92f9-47a3-8cb6-5c20f2923494.png">

15. If you go to the case details, in the Currently active task Send Final Notification Letter task is created. 

<img width="827" alt="image" src="https://user-images.githubusercontent.com/110078646/222509939-67cc6914-9cab-45c3-b5c8-0b2a14ddeb53.png">

16. In the Currently active task, from the actions dropdown you need to select and click on Resend Final Notification Letter

<img width="827" alt="image" src="https://user-images.githubusercontent.com/110078646/222510468-82888daf-669b-45ad-afff-be43c92b7bdb.png">

<img width="827" alt="image" src="https://user-images.githubusercontent.com/110078646/222510697-82579ae0-82fa-4d5f-b5c2-965a5a116a31.png">

17. In the case timeline you can see that Send final notification letter is completed
<img width="828" alt="image" src="https://user-images.githubusercontent.com/110078646/222511011-6a419f1b-1d8c-4c18-b7f6-ad2accff3401.png">
